pkg/kubelet/kubelet.go

```diff
@@ -1635,7 +1635,7 @@ func (kl *Kubelet) syncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	}
 
 	// ensure the kubelet knows about referenced secrets or configmaps used by the pod
-	if !kl.podWorkers.IsPodTerminationRequested(pod.UID) {
+	if !kl.podWorkers.IsPodTerminationRequested(pod.UID) && kl.kubeClient != nil {
 		if kl.secretManager != nil {
```

Comment:
question, why `kl.secretManager` and `kl.configMapManager` should be not nil in a standalone kubelet?

Reply:
It seems to be better to not set secret and configmap manager if `kubeClient` is nil.

Reply:
yep, no one to talk to :)